app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsTable/index.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+    useEffect,
+    useMemo,
+} from 'react';
+import { useOutletContext } from 'react-router-dom';
+import {
+    Container,
+    Pager,
+    Table,
+} from '@ifrc-go/ui';
+import { useTranslation } from '@ifrc-go/ui/hooks';
+import {
+    createBooleanColumn,
+    createElementColumn,
+    createStringColumn,
+    numericIdSelector,
+} from '@ifrc-go/ui/utils';
+import {
+    isDefined,
+    isNotDefined,
+} from '@togglecorp/fujs';
+
+import useFilterState from '#hooks/useFilterState';
+import { type CountryOutletContext } from '#utils/outletContext';
+import {
+    type GoApiResponse,
+    useRequest,
+} from '#utils/restRequest';
+
+import { VALIDATED } from '../common';
+import LocalUnitsTableActions, { type Props as LocalUnitsTableActionsProps } from './LocalUnitTableActions';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+const PAGE_SIZE = 15;
+
+type LocalUnitsTableResponse = GoApiResponse<'/api/v2/local-units/'>;
+type LocalUnitsTableListItem = NonNullable<LocalUnitsTableResponse['results']>[number];
+
+interface Props {
+    filter: {
+        type?: number;
+        search?: string;
+        isValidated?: string;
+    }
+}
+
+function LocalUnitsTable(props: Props) {
+    const {
+        filter: filterFromProps,
+    } = props;
+
+    const strings = useTranslation(i18n);
+    const { countryResponse } = useOutletContext<CountryOutletContext>();
+
+    const {
+        limit,
+        offset,
+        page,
+        setPage,
+        filtered,
+        filter,
+        setFilter,
+    } = useFilterState({
+        filter: {
+            ...filterFromProps,
+        },
+        pageSize: PAGE_SIZE,
+    });
+
+    useEffect(() => {
+        if (filterFromProps) {
+            setFilter(filterFromProps);
+        }
+    }, [filterFromProps, setFilter]);
```

Comment:
Is filter supposed to change after the component is mounted?
If possible can we not mount this component till the filter outside is loaded and then initialize the filter state as done above to avoid the useEffect?

Reply:
Yes, the filter can change after it is mounted. The initialization part is not an issue but we need to set the filter and reset the pagination as well. So, we have used the `useEffect` hook here.